src/views/view.c

```diff
@@ -915,18 +915,45 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
       PangoRectangle ink;
       PangoFontDescription *desc = pango_font_description_copy_static(darktable.bauhaus->pango_font_desc);
       pango_font_description_set_weight(desc, PANGO_WEIGHT_BOLD);
-      const int fontsize = 0.25 * width;
+      const int fontsize = 0.20 * width;
       pango_font_description_set_absolute_size(desc, fontsize * PANGO_SCALE);
       layout = pango_cairo_create_layout(cr);
       pango_layout_set_font_description(layout, desc);
       const char *ext = img->filename + strlen(img->filename);
       while(ext > img->filename && *ext != '.') ext--;
       ext++;
       cairo_set_source_rgb(cr, fontcol, fontcol, fontcol);
-      pango_layout_set_text(layout, ext, -1);
-      pango_layout_get_pixel_extents(layout, &ink, NULL);
-      cairo_move_to(cr, .025 * width - ink.x, .24 * height - fontsize);
-      pango_cairo_show_layout(cr, layout);
+
+
+      const int isportrait = ((img->orientation & ORIENTATION_SWAP_XY) != 0) != (img->height > img->width);
```

Comment:
this is not correct as it checks the original orientation. take a landscape picture, use crop&rotate and create a portrait composition. You'll see that the format will be hidden under the picture. You should use the height and width as cropped.

Reply:
Thanks, you're right.
I've been trying to figure out img->crop_width and img->crop_height, but both are always 0. My new commit works for me.